### PR TITLE
adds workflow for automatic merge v1 into main

### PR DESCRIPTION
**Pull request by ivansenic**
**What this PR does**:

Adds workflow that merges v1 into main and creates a PR. Runs on `stargate-v1-release` event or manual trigger.

**Which issue(s) this PR fixes**:
Internal

### CODE REVIEW

File: .github/workflows/merge-main-v1.yaml
Comment (jeffreyscarpenter) [inline .github/workflows/merge-main-v1.yaml:36]: Do we need to make this branch name different each time? In most cases the branch gets deleted once the PR is merged, but what if we don't merge the PR quickly enough for some reason?

Comment (ivansenic) [reply .github/workflows/merge-main-v1.yaml:36]: it is changed, it's gonna be `create-pull-request/merge-main-v1-aaabbb`, as `branch-suffix: short-commit-hash` defines.

Comment (jeffreyscarpenter) [reply .github/workflows/merge-main-v1.yaml:36]: interesting, thanks.

**Review by jeffreyscarpenter — APPROVED** [review]

**Pull request merged**
